addons/website_slides/models/slide_channel.py

```diff
@@ -462,11 +568,15 @@ def _compute_can_upload(self):
     def _compute_can_publish(self):
         """ For channels of type 'training', only the responsible (see user_id field) can publish slides.
         The 'sudo' user needs to be handled because they are the one used for uploads done on the front-end when the
-        logged in user is not publisher but fulfills the upload_group_ids condition. """
+        logged in user is not publisher but fulfills the upload_group_ids condition. Invited attendees can
+        preview the course as public and sudo. Prevent them from uploading."""
+        if self.env.is_superuser() and not self.env.user._is_public():
```

Comment:
Would be fun to have a public superuser :D ... not sure second part is really necessary (but it is not false, I agree).